core/src/test/java/org/apache/iceberg/TestSnapshotJson.java

```diff
@@ -49,12 +50,23 @@ public void testJsonConversion() throws IOException {
     String json = SnapshotParser.toJson(expected);
     Snapshot snapshot = SnapshotParser.fromJson(json);
 
-    Assert.assertEquals("Snapshot ID should match", expected.snapshotId(), snapshot.snapshotId());
-    Assert.assertEquals(
-        "Files should match", expected.allManifests(ops.io()), snapshot.allManifests(ops.io()));
-    Assert.assertNull("Operation should be null", snapshot.operation());
-    Assert.assertNull("Summary should be null", snapshot.summary());
-    Assert.assertEquals("Schema ID should match", Integer.valueOf(1), snapshot.schemaId());
+    assertThat(snapshot.snapshotId()).isEqualTo(expected.snapshotId());
+    assertThat(snapshot.allManifests(ops.io())).hasSize(2);
+
+    for (int i = 0; i < snapshot.allManifests(ops.io()).size(); i++) {
```

Comment:
are these new checks? can you elaborate why those are being added?

Reply:
@nastra I tried to check each element in the manifest list, but it's a bit redundant. I believe it's enough to check with `isEqualTo` between two lists. Let me fix this part and other part like this.